src/gui/application.cpp

```diff
@@ -489,17 +489,18 @@ void Application::slotownCloudWizardDone(int res)
         _checkConnectionTimer.start();
         slotCheckConnection();
 
-        // If one account is configured: enable autostart
-#ifndef QT_DEBUG
-        bool shouldSetAutoStart = (accountMan->accounts().size() == 1);
-#else
-        bool shouldSetAutoStart = false;
-#endif
+        // If at least one account is configured: enable autostart
+        bool shouldSetAutoStart = (accountMan->accounts().size() >= 1);
+
+        // Don't autostart development/test builds
+        shouldSetAutoStart = shouldSetAutoStart && !DEV_BUILD;
```

Comment:
Why did you not the one liner? With that `shouldSetAutoStart` can be const? :) If you feel like it, you can also comment that one liner.

Reply:
The reason I didn't set it as a `const` was that the number of accounts is not a constant, and anyway there is another assignment statement immediately following:
https://github.com/nextcloud/desktop/blob/b89e0345d8b7c081a36382f520be2036d7390b06/src/gui/application.cpp#L498-L502
Am I misunderstanding how `const` works?

The main reason I didn't go with the one-liner is that I'm personally less fond of syntax that's unique to C++, but also (a) yes, it does let me add a separate comment for each step, and (b) the structure mirrors that of Lines 498-502.

Reply:
It does not matter if the number of accounts is not a const. const is a beautiful language construct, it means that the variable can not be reassigned. In general you should avoid non const variables when ever possible. It makes it so much easier to reason about the code. Having preprocessor statements with non-const variables makes it even more difficult to understand whats going on. You are right the special case for mac os makes it even more difficult. If we really want to keep that statement, I would do the following
```cpp
        const auto isAccountConfigured = accountMan->accounts().size() == 1;
#ifndef Q_OS_MAC
        const auto isInstalled = QCoreApplication::applicationDirPath().startsWith("/Applications/");
        const auto shouldSetAutoStart = DEV_BUILD ? false : isAccountConfigured && isInstalled;
#else
        const auto shouldSetAutoStart = DEV_BUILD ? false : isAccountConfigured;
#endif
```
But I would say we can drop the special case for macos because we now have the dev build option that takes care of not autostarting the application. It makes no sense to me to have a special case just for macos.

The ternary operator `?:` is nothing unique to C++. It's also in C, Java, C#, Kotlin and JavaScript and maybe some other languages that I don't know. If you want you can comment that statement and if you want to make it a bit more readable you could add a variable like `isAccountConfigured`.

Reply:
Probably my biggest hesitation comes from the fact that the `const` keyword wasn't there before, and I don't want to accidentally break anything by adding it unnecessarily. I mean, I can use it; I just don't really want to have to deal with testing it.

FWIW there's a method available, `Utility::isMac()`, that doesn't require the use of header macros here, so I could do a single assignment statement if I use that.

I think my bias comes from the fact that I learned Java before I learned C, and Java is all about ridiculous verbosity, even if it still lets you use C-style abbreviated syntax in some places.